### ISSUE TITLE
Suggestion: Make interleave compatible with subheadings

### ISSUE THREAD

**Issue by Tobias-Loeffler** [rudolfochrist/interleave-mode#55]
Hey there!
I would love to see a better compatibility with the outline capabilities of org-mode, i.e. be able to take notes as sub-, subsub-, etc. headings. Lets assume we have a pdf structured as following:

```
1. Heading
some text
--pagebreak--
1.1 Subheading
more text
--pagebreak--
2. Another heading
and even more text
```
Then I would like to format the org-file with the notes as following:

```
* Heading (interleaved w/ p. 1)
  notes
** Subheading (interleaved w/ p. 2)
    notes
* Another heading (interleaved w/ p. 3)
  notes

```
If I format the headings file like that, there are two issues:
1) M-p and M-n don't switch properly between Headings and Subheadings, one "gets stuck on the lower level".
2) New notes are not stored in the proper place in between existing ones but appended at the end of the org-file.

Greetings and thanks for the otherwise great package :)
Tobias

**Comment by rudolfochrist** [comment]
As you pointed out this is currently not possible. `interleave` operates only on top-level headings (or one sub-level depth with multi-pdf notes). 

The problem is that you can't tell where to insert new notes. `interleave` isn't aware of any structure a given PDF might have. In other words we cannot figure out if (e.g.) page 4 belongs to section 1.2.4 or to section 2. This is the reason why new notes are appended at the end of the notes file. To add the notes to the _right_ location makes human interaction necessary.  But if you want to come up with a clever algorithm that figures out the _right_ location, I'll happily merge a PR 😄 

IRC <kbd>M-n</kbd>/<kbd>M-p</kbd> only searches along the siblings headings. Therefore you're stuck if a _lower_ level has been reached. The search can be expanded to search also in parent headings and the parent's siblings. This can collide with multi-pdf notes and implementing may be tricky. I image that some multi-pdf stuff has to be rewritten. But I can be mistaken on that point. 

To summarize:
1. Adding new notes to the _right_ location without prior knowledge of the structure of the PDF is next to impossible. It is indeed a very god suggestion, but won't be implemented anytime soon. If ever. 
2. Searching the parent heading (and siblings of the parent heading) for possible notes should be possible. And may be very good first contribution. 

Thanks you for your suggestions. 

-Sebastian

**Comment by weirdNox** [comment]
Hello there! I'm sorry to be hijacking this thread, but as this seems to be stagnant, I'll post about [org-noter](https://github.com/weirdNox/org-noter) here, which may help @Tobias-Loeffler (if you don't mind, of course, @rudolfochrist!).

If I understood correctly, this is supported in `org-noter`, where you may create hierarchies of notes, even out of order if you want, and then it will fetch them all when you are on their page.

As `org-noter` was inspired by Interleave, the core functionality is the same. Most of the features present in this package are also there, and it now has a lot of new features.

If you make the switch, [check this](https://github.com/weirdNox/org-noter#diff) in order to know what differences there are and the changes you must do in order to use it. Also, you may discover ways to customize `org-noter` by checking the [customization group](https://www.gnu.org/software/emacs/manual/html_node/emacs/Easy-Customization.html).

**Comment by rudolfochrist** [comment]
@weirdNox  Please feel free to do so :+1: . I don't think I will ever find the time to implement all the new features. Honestly, since I've graduated I really don't use `interleave-mode` myself anymore and I really lost the interest. Perhaps I add a note on top of the readme pointing to `org-noter` and tell people to use this instead of `interleave-mode`, because it seems to supersede it and is actively maintained. What do you think?

**Comment by weirdNox** [comment]
If you think `org-noter` is up to the task, that would be great, thanks :) 

> Honestly, since I've graduated I really don't use interleave-mode myself anymore and I really lost the interest.

Yeah, I made `org-noter` because it is something I need for studying everyday, but I bet that if I didn't use it, then I would probably lose interest too... We don't have time for everything, so decisions must be made :P

`interleave-mode` was a great start to something I would probably never think of, so I'm very grateful that you created it and made it public in the first place!
